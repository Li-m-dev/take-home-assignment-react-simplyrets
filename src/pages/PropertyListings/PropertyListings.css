.list_container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

@media screen and (min-width: 768px) {
  .list_container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-evenly;
    align-items: baseline;
  }
}
