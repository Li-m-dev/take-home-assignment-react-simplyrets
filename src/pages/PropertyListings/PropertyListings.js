import PropertyCard from 'components/PropertyCard/PropertyCard';
import React, { useState, useEffect } from 'react';
import './PropertyListings.css';

const PropertyListings = () => {
  const [propertyListings, setPropertyListings] = useState([]);
  const saved = localStorage.getItem('favoriteList');
  const [favList, setFavList] = useState(JSON.parse(saved) || []);

  useEffect(() => {
    getAllProperties();
  }, []);

  const getAllProperties = async () => {
    const config = {
      headers: new Headers({
        Authorization: 'Basic ' + btoa('simplyrets:simplyrets'),
      }),
    };
    try {
      const response = await fetch(
        'https://api.simplyrets.com/properties?status=Active',
        config,
      );
      const data = await response.json();
      setPropertyListings(data);
    } catch (error) {
      throw new Error(`fetch data fail with error message: ${error.message}`);
    }
  };

  const toggleFav = (mlsId) => {
    let favListCopy = [...favList];

    if (favList.includes(mlsId)) {
      favListCopy = favListCopy.filter((id) => id !== mlsId);
    } else {
      favListCopy.push(mlsId);
    }

    localStorage.setItem('favoriteList', JSON.stringify(favListCopy));
    setFavList(favListCopy);
  };

  const propertiesToDisplay = propertyListings.map((listedProperty) => {
    const { address, photos, property, listPrice, listingId, listDate, mlsId } =
      listedProperty;
    return (
      <PropertyCard
        key={listingId}
        address={address}
        photos={photos}
        property={property}
        listPrice={listPrice}
        listDate={listDate}
        mlsId={mlsId}
        isSavedFav={favList.includes(mlsId)}
        onToggleFav={toggleFav}
      />
    );
  });

  return <div className="list_container">{propertiesToDisplay}</div>;
};

export default PropertyListings;
