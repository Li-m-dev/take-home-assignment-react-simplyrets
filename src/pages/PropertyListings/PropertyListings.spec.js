import React from 'react';
import { render } from '@testing-library/react';
import PropertyListings from './PropertyListings';

describe('PropertyListings tests', () => {
  it('should render property lists', () => {
    // Arrange
    // Act
    const { container } = render(<PropertyListings />);

    // Assert
    expect(container).toMatchSnapshot();
  });

  //TODO: Add more tests here
});
