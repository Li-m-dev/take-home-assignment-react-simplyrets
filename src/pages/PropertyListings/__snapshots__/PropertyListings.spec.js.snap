// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyListings tests should render property lists 1`] = `
<div>
  <div
    class="list_container"
  />
</div>
`;
